src/js/oncoprintlegendrenderer.js

```diff
@@ -154,9 +154,11 @@ var OncoprintLegendView = (function() {
 	var config = rule.getLegendConfig();
 	if (config.type === 'rule') {
 	    var concrete_shapes = rule.apply(config.target, model.getCellWidth(true), view.base_height);
-	    // generate backgrounds
-	    root.appendChild(svgfactory.bgrect(6, 20, 'rgb(190,190,190)'));
-	    // generate shapes
+            if (config.baseLegendShapes) {
+                // generate backgrounds
+                root.appendChild(svgfactory.bgrect(6, 20, 'rgb(190,190,190)'));
```

Comment:
there was a misunderstanding - the idea is that in the rule we'd specify the particular baseLegendShapes, so baseLegendShapes is of type Shape[] not boolean

Reply:
if we're going to do it this way, which I think actually does make sense, we should change the name of baseLegendShapes to baseGrayRect or something

Reply:
will go for the 2nd option

Reply:
btw, please use `svgfactory.rect`, and use `model.getCellWidth(true)` and `view.base_height` instead of hardcoded 6,20

Reply:
why `rect' not `bgrect'?

Reply:
bgrect is just semantically for the white background. I mean its basically the same, just semantically thats not what it's meant for so I think it's cleaner to stick with rect.